### DISCUSSION TITLE
Is TurboPack using SWC under the hood ?

### DISCUSSION THREAD

**Discussion by zakuru** [vercel/turborepo#2350] [Help] [answered]
Hello,

I would like to know if TurboPack  is based on SWC compiler and features ?
If so, what are the additional features that TurboPack  adds ?

Thanks.

**Comment by jridgewell** [comment, marked as answer]
Turbopack uses SWC to parse, transpile, and print JS/CSS/JSON, etc. In the future, we'll use it to minify production outputs.

Turbopack provides bundling, a dev server, a hot-module reload runtime, etc.

**Reply by bruce-c-liu** [reply]
Followup: Does Turbopack use spack (swcpack)?

**Reply by jridgewell** [reply]
No, we're implementing our own bundling

**Reply by DNA-PC** [reply]
@jridgewell I'm looking for an alternative to SWC Spack.
[From their page](https://swc.rs/docs/usage/bundling), they're basically saying Spack won't be actively developed further.

I'm seeking to configure multiple entrypoints, that will generate different scripts, and build all of that through a single command.
This is how I did it using Spack:
```js
const fs = require('fs');
const { config } = require('@swc/core/spack');

// Use the same options for spack as the ones in the .swcrc file (better intellisense there)
const options = JSON.parse(fs.readFileSync('.swcrc', { encoding: 'utf8', flag: 'r' }));

module.exports = config({
  entry: {
    'xxx-gs': `${__dirname}/src/apps/xxx/global-script.ts`,
    'xxx-pps': `${__dirname}/src/apps/xxx/per-page-script.ts`,
  },
  output: {
    path: __dirname + '/dist',
  },
  module: {},
  mode: 'production',
  target: 'browser',
  options,
});
```

Can Turbo help me with this?

The main reason I want to get away from Spack, is that it doesn't handle async/await. (build crashes)
